artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -1029,6 +1131,25 @@ public boolean page(Message message,
          lock.readLock().unlock();
       }
 
+      if (pageFull && pageFullMessagePolicy != null) {
```

Comment:
Related to earlier comment, if it validates you have a policy, can this then just simply check "if (pageFull)" ?

Reply:
Probably. It feels wrong not doing the redundant check though! :)

Reply:
My OCD wants to keep it, but I will remove it :)